pkg/virt-api/rest/cpu.go

```diff
@@ -0,0 +1,70 @@
+package rest
+
+import (
+	"context"
+	v1 "kubevirt.io/api/core/v1"
+	"kubevirt.io/kubevirt/pkg/apimachinery/patch"
+	"net/http"
+	"strconv"
+
+	"github.com/emicklei/go-restful/v3"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+
+	"kubevirt.io/kubevirt/pkg/virt-api/definitions"
+)
+
+func (app *SubresourceAPIApp) CPURequestHandler(request *restful.Request, response *restful.Response) {
+	namespace := request.PathParameter(definitions.NamespaceParamName)
+	name := request.PathParameter(definitions.NameParamName)
+	cpuCountStr := request.QueryParameter(definitions.CpuCountParamName)
+
+	if cpuCountStr == "" {
+		writeError(errors.NewBadRequest("cpuCount parameter is required"), response)
+		return
+	}
+
+	cpuCount, err := strconv.Atoi(cpuCountStr)
+	if err != nil {
+		writeError(errors.NewBadRequest("cpuCount parameter must be an integer"), response)
+		return
+	}
+
+	if cpuCount < 0 {
+		writeError(errors.NewBadRequest("cpuCount parameter must be non-negative"), response)
+		return
+	}
+
+	ctx, cancel := context.WithTimeout(context.Background(), requestTimeout)
+	defer cancel()
+
+	// Validate that the VM exists
+	if _, err = app.virtCli.VirtualMachine(namespace).Get(ctx, name, metav1.GetOptions{}); err != nil {
+		if errors.IsNotFound(err) {
+			writeError(errors.NewNotFound(v1.Resource("virtualmachine"), name), response)
+		} else {
+			writeError(errors.NewInternalError(err), response)
+		}
+		return
+	}
+
+	patchOperation := patch.PatchOperation{
+		Op:    patch.PatchReplaceOp,
+		Path:  "/spec/template/spec/domain/cpu/cores",
+		Value: cpuCount,
+	}
+
+	patchPayload, err := patch.GeneratePatchPayload(patchOperation)
```

Comment:
```suggestion
	patchPayload, err := patch.New(
	    patch.WithReplace( "/spec/template/spec/domain/cpu/cores", cpuCount),
	).GeneratePayload()
```

You would probably also need a test for the old value to not overwriting a newer change